selenium/spec/asset_library_urls_spec.rb

```diff
@@ -0,0 +1,123 @@
+require_relative 'spec_helper'
+
+describe 'Adding Asset Library URLs' do
+
+  test_id = WebDriverUtils.test_course_name(self)
+  test_users = WebDriverUtils.load_test_users
+  test_teacher = test_users['Teacher 1']
+
+  before(:all) { @driver = WebDriverUtils.driver }
+
+  before(:context) do
+    @canvas = CanvasPage.new @driver
+    @canvas.load_homepage
+    @cal_net= CalNetPage.new @driver
+    @cal_net.log_in(WebDriverUtils.admin_username, WebDriverUtils.admin_password)
+    @course_id = @canvas.create_complete_test_course(test_id, test_users)
+    @canvas.log_out
+    @cal_net.logout_success_message_element.when_visible WebDriverUtils.page_load_wait
+    @canvas.load_homepage
+    @cal_net.log_in(test_teacher['username'], WebDriverUtils.test_user_password)
+    @canvas.load_course_site @course_id
+    @canvas.accept_login_messages @course_id
+    @asset_library = AssetLibraryPage.new @driver
+    @asset_library_url = @canvas.click_asset_library_link
+    @asset_library.load_page(@driver, @asset_library_url)
+    @asset_library.click_manage_categories_link
+    @asset_library.add_category 'Category 1'
+    @asset_library.add_category 'Category 2'
+    @asset_library.delete_category(@driver, 1)
+  end
+
+  before(:example) do
+    @asset_library.load_page(@driver, @asset_library_url)
+    @asset_library.click_add_site_link
+  end
+
+  it 'allows the user to add a URL, title, category, and description to the asset library' do
+    @asset_library.enter_url_metadata('en.wikipedia.org', 'URL Title 1', 'Category 1', 'URL description')
```

Comment:
If this is an action that'll be taking place a lot, you might want to wrap these 2 functions into a "create_url" function that automatically generates some metadata and adds the link to the asset library